### PR TITLE
Second batch of improvements from enterprise app

### PR DESCRIPTION
**Pull request by e-kolpakov**
**Description:** This PR contains fixes and improvements found working on new edx-enterprise app.

**JIRA:** [ENT-21](https://openedx.atlassian.net/browse/ENT-48)

**Related:** https://github.com/edx/edx-enterprise/pull/8

**Testing instructions:**

Run tests and make sure they are passing.
Try generating new app (as described in this repo's README) and check the following:

- generated docs/{{appname}}.*.rst is gitignored (add a package and run `make docs`)
- `make diff_cover` runs test suite and reports diff coverage.
- `make extract_translations` after `make docs` does not generate djangojs.po with strings from `docs/_build`
- Repo contains PR template
- codecov config file is present
- Google style docstrings are properly parsed by Sphinx (i.e. `Arguments` clause is parsed into parameters, see https://github.com/edx/edx-enterprise/pull/8#discussion_r87990254)
- `make quality` does not produce pycodestyle warning about `pep8` section in tox.ini

**Reviewers:**
- [x] @mtyaka (OpenCraft internal review)
- [x] @jmbowman

### CODE REVIEW

**Comment by e-kolpakov** [comment]
@mtyaka @jmbowman please review

File: tox.ini [hunk 1]
```diff
@@ -2,7 +2,7 @@
 envlist = py{27,35}
 skipsdist = true
 
-[pep8]
+[pycodestyle]
```
Comment (jmbowman) [inline tox.ini]: Cool, looks like they just released the fix for this 2 weeks ago.  I tried this right after the pep8 -> pycodestyle rename, but the code didn't support it yet.

File: {{cookiecutter.repo_name}}/.gitignore [hunk 2]
```diff
@@ -61,6 +61,7 @@ output/*/index.html
 docs/_build
 docs/modules.rst
 docs/{{ cookiecutter.app_name }}.rst
+docs/docs/{{ cookiecutter.app_name }}.*.rst
```
Comment (jmbowman) [inline {{cookiecutter.repo_name}}/.gitignore]: Why would anything get created under ``docs/docs``?  That looks like there was some kind of mistake in the Sphinx configuration

File: {{cookiecutter.repo_name}}/README.rst [hunk 3]
```diff
@@ -62,6 +62,9 @@ Please read `How To Contribute <https://github.com/edx/edx-platform/blob/master/
 Even though they were written with ``edx-platform`` in mind, the guidelines
 should be followed for Open edX code in general.
 
+PR description template can be found at
+`PR_TEMPLATE.md <https://{{ cookiecutter.repo_name }}/blob/master/PR_TEMPLATE.md>`_
```
Comment (mtyaka) [inline {{cookiecutter.repo_name}}/README.rst]: Should this be `https://github.com/edx/{{ cookiecutter.repo_name }}/blob/master/PR_TEMPLATE.md`?

**Pull request merged**